python/python/bystro/proteomics/listener_annotation_interface.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+
+import logging
+from ruamel.yaml import YAML
+from pathlib import Path
+import pyarrow.dataset as ds
+from opensearchpy import OpenSearch
+
+from bystro.beanstalkd.worker import listen, QueueConf, ProgressPublisher
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.proteomics.annotation_interface import (
+    get_annotation_result_from_query,
+    join_annotation_result_to_proteomics_dataset,
+)
+
+logger = logging.getLogger(__file__)
+
+EFS_BASE_PATH = "/seqant/user-data/"
+PROTEOMICS_TUBE = "proteomics"
+
+
+class ProteomicsJobData(BaseMessage, frozen=True, rename="camel"):
+    """
+    The expected JSON message for the Proteomics job.
+
+    Parameters
+    ----------
+    submission_id: str
+        The unique identifier for the job.
+    data_path: str
+        The path to the proteomics data file.
+    out_dir: str
+        The directory to write the results to.
+    annotation_query: str
+        The query string used to fetch annotation data from OpenSearch index.
+    index_name: str
+        The name of the OpenSearch index from which annotation data will be queried.
+    """
+
+    submission_id: str
+    data_path: str
+    out_dir: str
+    annotation_query: str
+    index_name: str
+
+
+class ProteomicsJobCompleteMessage(CompletedJobMessage, frozen=True, kw_only=True, rename="camel"):
+    result_path: str
+
+
+def _load_queue_conf(queue_conf_path: str) -> QueueConf:
+    with Path(queue_conf_path).open(encoding="utf-8") as queue_config_file:
+        raw_queue_conf = YAML(typ="safe").load(queue_config_file)
+    beanstalk_conf = raw_queue_conf["beanstalkd"]
+    return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
+
+
+def handler_fn(_publisher: ProgressPublisher, job_data: ProteomicsJobData) -> str:
+    logger.info("Processing Proteomics job: %s", job_data)
+
+    efs_out_dir = Path(EFS_BASE_PATH) / job_data.out_dir
```

Comment:
You don't need this. job_data.out_dir is a fully qualified path to the output.

We also can't assume job_data.out_dir refers to an EFS path; it could be an S3 or google cloud bucket path in the future.